packages/api/src/middleware/validation/checkMarketDirectoryInfo.ts

```diff
@@ -36,114 +37,157 @@ export interface MarketDirectoryInfoRouteParams {
+          ...partner,
+          market_directory_info_id: id, // needed as joint primary key
+          deleted: false, // un-delete if previously deleted
+        }));
+      }
 
-    for (const emailProperty of ['contact_email', 'email'] as const) {
-      if (req.body[emailProperty] && !isValidEmail(req.body[emailProperty])) {
-        return res.status(400).send(`Invalid ${emailProperty}`);
+      for (const emailProperty of ['contact_email', 'email'] as const) {
+        if (req.body[emailProperty] && !isValidEmail(req.body[emailProperty])) {
+          return res.status(400).send(`Invalid ${emailProperty}`);
+        }
       }
-    }
 
-    if (address && !(await isValidAddress(address))) {
-      return res.status(400).send('Invalid address');
-    }
+      if (address && !(await isValidAddress(address))) {
+        return res.status(400).send('Invalid address');
+      }
 
-    if (website && !(await isValidUrl(website))) {
-      return res.status(400).send('Invalid website');
-    }
+      if (website && !(await isValidUrl(website))) {
+        return res.status(400).send('Invalid website');
+      }
 
-    for (const social of SOCIALS.filter((social) => req.body[social]?.trim())) {
-      const socialUsername = validateSocialAndExtractUsername(social, req.body[social]!);
+      for (const social of SOCIALS.filter((social) => req.body[social]?.trim())) {
+        const socialUsername = validateSocialAndExtractUsername(social, req.body[social]!);
 
-      if (!socialUsername) {
-        return res.status(400).send(`Invalid ${social}`);
+        if (!socialUsername) {
+          return res.status(400).send(`Invalid ${social}`);
+        }
+        req.body[social] = socialUsername;
       }
-      req.body[social] = socialUsername;
-    }
 
-    next();
+      next();
+    } catch (error) {
+      console.error(error);
+      if (error instanceof LiteFarmCustomError) {
+        return error.body
+          ? res.status(error.code).json({ ...error.body, message: error.message })
+          : res.status(error.code).send({
+              title: error.message,
+              message: error.message,
+              status: error.code,
+              original: error.message,
+            });
+      }
+      return res.status(500).json({
+        error,
+      });
+    }
   };
 }
 
 export function checkMarketDirectoryInfoRecord(
   { errorMessage } = { errorMessage: 'Market directory info not found' },
 ) {
   return async (
-    req: LiteFarmRequest<unknown, unknown, unknown, unknown>,
+    req:
+      | LiteFarmRequest<unknown, MarketDirectoryInfoRouteParams, unknown, unknown>
+      | Request<MarketDirectoryInfoRouteParams, unknown, unknown, unknown>,
     res: Response,
     next: NextFunction,
   ) => {
-    // @ts-expect-error: TS doesn't see query() through softDelete HOC; safe at runtime
-    const record = await MarketDirectoryInfoModel.query().findById(req.params.id).whereNotDeleted();
```

Comment:
After adding the try/catch this line wraps with prettier and exposes a type error on req.params. 

This prevented compile and was fixed both temporarily in first commit, then a more robust fix second commit"